interfaces/ValuePoint.h

```diff
@@ -312,7 +316,27 @@ namespace Exchange {
     protected:
         inline void Updated() const
         {
-            _job.Submit();
+            if (_clients.size() > 0) {
+                notify metadata = notify::METADATA;
+                notify none     = notify::NONE;
+
+                if ((_notify.compare_exchange_strong(none, notify::UPDATE) == true) ||
+                    (_notify.compare_exchange_strong(metadata, static_cast<notify>(notify::METADATA | notify::UPDATE)) == true)) {
+                    _job.Submit();
+                }
+            }
+        }
+        inline void Metadata() const
+        {
+            if (_clients.size() > 0) {
+                notify update = notify::UPDATE;
+                notify none   = notify::NONE;
+
+                if ((_notify.compare_exchange_strong(none, notify::METADATA) == true) ||
+                    (_notify.compare_exchange_strong(update, static_cast<notify>(notify::METADATA | notify::UPDATE)) == true)) {
+                    _job.Submit();
```

Comment:
Perhaps I'm missing something but in case the other reason to submit took place you now submit the job for the 2nd time (suppose an Update() just was called then now the compare(none,...) will fail, compare(upadate,.. will succeed indeed setting the mode to both bit submitting the job for the 2nd time, which I guess is not what you want)

Reply:
Indeed it will submit the job again. It is a waste of the second run as during the first run it will process both notifications. Lets see if we could avoid the second run...